hw2/1_mini_logo.hs

```diff
@@ -1 +1,89 @@
 -- Mini Logo
+
+-- Mini Logo is a simplified version of the Logo language for programming
+-- 2D graphics. The idea behind Logo and Mini Logo is to describe simple
+-- line graphics through commands to move a pen from one position to
+-- another. The pen can either be "up" or "down". Positions are given
+-- by pairs of integers. Functions can be defined (using def) and called
+-- (using call) to reuse groups of commands. The syntax of Mini Logo is as
+-- follows
+
+-- part a: define the abstract syntax for mini logo as Cmd
+
+data Mode = Up | Down deriving (Show)
+
+data Coord = Ints (Int, Int) | Names (String, String) deriving (Show)
+
+data Pars = Pars String Pars | Name String deriving (Show)
+
+data Vals = Vals Int Vals | Int deriving (Show)
+
+data Cmd
+  = Pen Mode
+  | MoveTo Coord
+  | Def String Pars Cmd
+  | Call Vals String
+  | Seq Cmd Cmd
+  deriving (Show)
+
+seqCmds :: [Cmd] -> Cmd
```

Comment:
I appreciate the readability this offers when you use it in the vector definition below. We could add a similar thing for Pars; I'd be happy to do that after this is merged.

Reply:
I like that, I'll leave it to you